### PR TITLE
Functions for obtaining code from image and separate deploy params

### PR DESCRIPTION
**Pull request by AlexeyVavilin**

### CODE REVIEW

File: ton_client/client/src/contracts/deploy.rs [hunk 1]
```diff
@@ -124,6 +153,80 @@ pub(crate) fn encode_message(_context: &mut ClientContext, params: ParamsOfDeplo
...
+        .map_err(|err| ApiError::contracts_deploy_image_creation_failed(err))?;
+
+    debug!("<-");
+    Ok(ResultOfGetCodeFromImage {
+        codeBase64: base64::encode(&image.get_serialized_code()
+            .map_err(|err| ApiError::contracts_deploy_image_creation_failed(err))?),
+    })
+}
+
+pub(crate) fn get_deploy_data(_context: &mut ClientContext, params: ParamsOfGetDeployData) -> ApiResult<ResultOfGetDeployData> {
```
Comment (Keshoid) [inline ton_client/client/src/contracts/deploy.rs]: A lot of work in one function. Can it be splitted to 2 parts? update image data and return data from image.

File: ton_client/client/src/contracts/mod.rs [hunk 2]
```diff
@@ -65,6 +65,10 @@ pub(crate) fn register(handlers: &mut DispatchTable) {
         deploy::encode_unsigned_message);
     handlers.spawn("contracts.deploy.address",
         deploy::get_address);
+    handlers.spawn("contracts.deploy.code_from_image",
```
Comment (Keshoid) [inline ton_client/client/src/contracts/mod.rs]: Why it is in deploy namespace? It's more general feature.

File: ton_client/client/src/contracts/run.rs [hunk 3]
```diff
@@ -253,14 +284,43 @@ pub(crate) fn decode_unknown_output(_context: &mut ClientContext, params: Params
...
+        params.internal)
+            .map_err(|err|ApiError::contracts_decode_run_output_failed(err))?;
     Ok(ResultOfDecodeUnknownRun {
         function: result.function_name,
         output: serde_json::from_str(result.params.as_str())
             .map_err(|err| ApiError::contracts_decode_run_output_failed(err))?
     })
 }
 
+pub(crate) fn get_run_body(_context: &mut ClientContext, params: ParamsOfGetRunBody) -> ApiResult<ResultOfGetRunBody> {
```
Comment (Keshoid) [inline ton_client/client/src/contracts/run.rs]: Function name is strange:) Can you call it like build_message_body or create_message_body or build_run_body?

File: ton_sdk/src/contract.rs [hunk 4]
```diff
@@ -106,28 +106,86 @@ impl ContractImage {
         Ok(Self{ state_init, id })
     }
 
-    pub fn from_state_init_and_key<T>(state_init_bag: &mut T, pub_key: &PublicKey) -> SdkResult<Self>
+    pub fn new_empty() -> SdkResult<Self> {
```
Comment (Keshoid) [inline ton_sdk/src/contract.rs]: The name is not like in rust style guide. Use new().

File: ton_client/client/src/contracts/deploy.rs [hunk 5]
```diff
@@ -124,6 +153,80 @@ pub(crate) fn encode_message(_context: &mut ClientContext, params: ParamsOfDeplo
...
 }
 
+pub(crate) fn get_code_from_image(_context: &mut ClientContext, params: ParamsOfGetCodeFromImage) -> ApiResult<ResultOfGetCodeFromImage> {
+    debug!("-> contracts.deploy.image.code()");
+
+    let bytes = base64::decode(&params.imageBase64)
+        .map_err(|err| ApiError::contracts_deploy_invalid_image(err))?;
+    let mut reader = Cursor::new(bytes);
+    let image = ContractImage::from_state_init(&mut reader)
+        .map_err(|err| ApiError::contracts_deploy_image_creation_failed(err))?;
```
[the remark this answers is not in the document]
Comment (AlexeyVavilin) [reply ton_client/client/src/contracts/deploy.rs]: It is not `deploy image`. It is contracts.deploy.image_creation_failed

File: ton_client/client/src/contracts/deploy.rs (hunk 1 above)
Comment (AlexeyVavilin) [reply ton_client/client/src/contracts/deploy.rs]: I don't think it is needed. It's splitting just for splitting

**Pull request merged**